src/test/java/io/strimzi/kafka/bridge/http/ProducerIT.java

```diff
@@ -1117,4 +1122,94 @@ void sendSimpleMessageWithWrongContentType(VertxTestContext context) throws Inte
+
+        JsonObject record = createDynamicRecord(key, value, firstContentType);
+        JsonArray records = new JsonArray().add(record);
+        JsonObject root = new JsonObject().put("records", records);
+
+        LOGGER.info("Sending:\n{}", root.toString());
+
+        // Send the first request
+        producerService()
+            .sendRecordsRequest(topic, root, firstContentType)
+            .sendJsonObject(root, response -> {
+                if (response.succeeded()) {
+                    HttpResponse<JsonObject> httpResponse = response.result();
+                    context.verify(() -> {
+                        assertThat(httpResponse.statusCode(), is(HttpResponseStatus.OK.code()));
+                        LOGGER.info("Successfully processed record with content type: {}", secondContentType);
+                    });
+                } else {
+                    context.failNow(response.cause());
+                }
+            });
+
+        record = createDynamicRecord(key, value, secondContentType);
+        records = new JsonArray().add(record);
+        root = new JsonObject().put("records", records);
+
+        // Send the second request
+        producerService()
+            .sendRecordsRequest(topic, root, secondContentType)
+            .sendJsonObject(root, response -> {
+                if (response.succeeded()) {
+                    HttpResponse<JsonObject> httpResponse = response.result();
+                    context.verify(() -> {
+                        assertThat(httpResponse.statusCode(), is(HttpResponseStatus.OK.code()));
+                        LOGGER.info("Successfully processed record with content type: {}", secondContentType);
+                        context.completeNow();
+                    });
+                } else {
+                    context.failNow(response.cause());
+                }
+            });
+
+        assertThat(context.awaitCompletion(TEST_TIMEOUT, TimeUnit.SECONDS), is(true));
+    }
+
+    private JsonObject createDynamicRecord(JsonObject key, JsonObject value, String contentType) {
+        JsonObject record = new JsonObject();
+
+        switch (contentType) {
+            case BridgeContentType.KAFKA_JSON_JSON:
+                // Key and value are already JSON objects
+                record.put("key", key);
+                record.put("value", value);
+                break;
+            case BridgeContentType.KAFKA_JSON_BINARY:
+                record.put("key", Base64.getEncoder().encodeToString(key.encode().getBytes()));
+                record.put("value", Base64.getEncoder().encodeToString(value.encode().getBytes()));
+                break;
+            case BridgeContentType.KAFKA_JSON_TEXT:
+                record.put("key", "key"); // Use pure plain text for the key
+                record.put("value", "value"); // Use pure plain text for the value
```

Comment:
but then it's hard coded this way? If you pass a string through a JsonObject, can't you get the raw value without the quotes? Asking ... needs to be verified.

Reply:
I can use 
```java
key.getString("id");
value.getString("description");
```
but that way it would be test-specific based on the JsonObject you provide. 

It would work in test case where we provide:
```java
final JsonObject key = new JsonObject().put("id", 123);
final JsonObject value = new JsonObject().put("id", 10).put("price", 150).put("description", "Hello world");
```
If you want this that way I am fine. But I didn't want to not rely on JSON objects inside the test...

Reply:
I see that, but hard coded would not be better. In the end I see the same problem for the `binary` type.
You are encoding in Base64 the JSON objects anyway .... but in reality they can be anything ... they are just binary not specifically JSON (in the end you are getting bytes from JSON). I am starting to think we cannot have one generic method but we need several ones for handling: JSON object for `json`, just bytes as binary for `binary` and plain string for `text`.